plugins/woocommerce/includes/admin/reports/class-wc-report-customer-list.php

```diff
@@ -263,28 +263,25 @@ public function prepare_items() {
 
 		add_action( 'pre_user_query', array( $this, 'order_by_last_name' ) );
 
-		/**
-		 * Get users.
-		 */
-		$admin_users = new WP_User_Query(
+		$privileged_users = new WP_User_Query(
 			array(
-				'role'   => 'administrator',
-				'fields' => 'ID',
-			)
-		);
-
-		$manager_users = new WP_User_Query(
-			array(
-				'role'   => 'shop_manager',
-				'fields' => 'ID',
-			)
+				'fields'     => 'ID',
```

Comment:
There are a couple issues with this.
1. It won't work exactly the same on some sites as the user_level doesn't always map directly to roles/capabilities.  Plugins can modify those.
2. The meta_key used won't always be correct for multisite instances.

Reply:
Gotcha, then we are better off using the built-in roles filter - we still improve the performance as we are running one quiry instead of two.

Reply:
Updated to use roles in db1610fc42113f537b25a25e0cf01fa41cfb6188